### ISSUE TITLE
Better naming schema for lambda functions

### ISSUE THREAD

**Issue by pauldhein** [ml4ai/delphi#186]
I would like the generated lambdas in the `*_lambdas.py` files to be changed from their current form:
```python
def petpt__lambda__td_0(tmax, tmin):
    return ((0.6*tmax)+(0.4*tmin))

def petpt__lambda__IF_1_0(xhlai):
    return (xhlai<=0.0)

def petpt__lambda__albedo_2(IF_1_0, albedo_0, albedo_1):
    return albedo_0 if IF_1_0 else albedo_1
```
to the following form:
```python
def petpt__assign__td_0(tmax, tmin):
    return ((0.6*tmax)+(0.4*tmin))

def petpt__condition__IF_1_0(xhlai):
    return (xhlai<=0.0)

def petpt__decision__albedo_2(IF_1_0, albedo_0, albedo_1):
    return albedo_0 if IF_1_0 else albedo_1
```
This will allow me to more quickly inspect the lambdas file by searching function names based on the type of function I am searching for. Not only that, but I believe we can condense the JSON specification for `__assign__`, `__condition__`, and `__decision__` function objects by using this format. Currently the outputted JSON for three functions mentioned above is:
```json
{
  "name": "petpt__condition__IF_1_0",
  "type": "condition",
  "target": "IF_1",
  "sources": [
    {
      "name": "xhlai",
      "type": "variable"
    }
  ],
  "body": [
    {
      "type": "lambda",
      "name": "petpt__lambda__IF_1_0",
      "reference": 17
    }
  ]
},
{
  "name": "petpt__assign__albedo_0",
  "type": "assign",
  "sources": [
    {
      "name": "msalb",
      "type": "variable"
    }
  ],
  "body": [
    {
      "type": "lambda",
      "name": "petpt__lambda__albedo_0",
      "reference": 18
    }
  ],
  "target": "albedo"
},
{
  "name": "petpt__decision__albedo_0",
  "type": "decision",
  "target": "albedo",
  "sources": [
    {
      "name": "IF_1_0",
      "type": "variable"
    },
    {
      "name": "albedo_3",
      "type": "variable"
    },
    {
      "name": "albedo_2",
      "type": "variable"
    }
  ]
},
```
and with my proposed format I believe we can condense the output into the following:
```json
{
  "name": "petpt__condition__IF_1_0",
  "type": "condition",
  "target": "IF_1",
  "reference": 17,
  "sources": [
    {
      "name": "xhlai",
      "type": "variable"
    }
  ],
},
{
  "name": "petpt__assign__albedo_0",
  "type": "assign",
  "reference": 18,
  "sources": [
    {
      "name": "msalb",
      "type": "variable"
    }
  ],
  "target": "albedo"
},
{
  "name": "petpt__decision__albedo_0",
  "type": "decision",
  "target": "albedo",
  "reference": 19,
  "sources": [
    {
      "name": "IF_1_0",
      "type": "variable"
    },
    {
      "name": "albedo_3",
      "type": "variable"
    },
    {
      "name": "albedo_2",
      "type": "variable"
    }
  ]
},
```

Will this representation work for everyone (@cl4yton and @adarshp)?

**Comment by cl4yton** [comment]
I agree, this looks like a good idea. If I follow, there are two things we're gaining: (1) the name of the function carries more about the type of function, and (2) there does not need to be a distinction between the name as it appears in the function_spec and then the lambda function (that currently is specified/reference in the function_spec body).

One thing I want to clarify: your example of the current and proposed json appears to drop the example of `petpt__assign__td_0`, replacing the `petpt__assign__td_0` with `petpt__assign__albedo_0` -- is that right? (I think it might just be a copy error, but want to make sure I'm not missing something.

I'll take a look at what needs to be updated in the GrFN spec that @pratikbhd needs to be aware of.

**Comment by adarshp** [comment]
This looks good to me!

**Comment by pauldhein** [comment]
@cl4yton Yes the example has a bit of a copypasta bug. I chose to use `petpt__lambda__td_0` as my example of an assignment statement in the lambdas file and I used `petpt__assign__albedo_0` as my example of an assignment function object in the json file. Sorry for the confusion, I did not intend to conflate those names!

**Comment by cl4yton** [comment]
@pauldhein cool, no worries, just wanted to verify.

I think the only things that could cause issues here is if there was a chance of name conflicts (that is, if there was a reason on the program analysis side for why we'd want to separate the GrFN-internal name (in the fn spec) from the name used in the Python lambda file.
@pratikbhd , @skdebray : what do you think?

**Comment by pauldhein** [comment]
Closed by #199
